source/Patches/Roles/RoleSingleton.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Linq;
+using Reactor;
+using static TownOfUs.Roles.Role;
+
+namespace TownOfUs.Roles
+{
+    public static class RoleSingleton<T> where T : Role
+    {
+        private static T _instance;
+        public static T Instance => _instance ??= RoleSingleton.OfType<T>().Single();
+
+        internal static void LoadSingletons()
+        {
+            foreach (var type in typeof(TownOfUs).Assembly.GetTypes())
+            {
+                if (!type.IsSubclassOf(typeof(Role))) continue; 
```

Comment:
Probably you’ll need checks here to make sure it’s concrete? Some of the stuff I have in flight adds I think adds subclasses of Role (to unify some of the handing of cooldowns).